### ISSUE TITLE
Editor loading very slowly

### ISSUE THREAD

**Issue by christad92** [OpenFn/lightning#2617]
Today during a demo I noticed that the editor was very slow when loading and had to wait for about one minute for it to load. The input and output panels were loaded but the editor panel was blank.  

Workflow:

**Comment by stuartc** [comment]
I can't reproduce this, has it happened subsequently? With my cache disabled I timed this as fully usable in 5 seconds.

**Comment by josephjclark** [comment]
I suspect this is related to cache and slow internet. There's like 6mb of editor stuff that needs downloading and if its not cached then it will take some time.

**Comment by josephjclark** [comment]
Huh - I just saw this!

I loaded [this workflow](https://app.openfn.org/projects/2fedca61-bb42-4e65-a908-7a60f1b25c1e/w/ce0cd63c-4aa1-4e5b-88be-813d125ebb8a?s=6637fdf1-62f4-446d-8f59-98b7b37dc3e3&a=2a30533e-0b31-4d5c-ba62-db49de5235ad&v=72&m=expand#log) on app.openfn.org, using a good fast stable connection and _surely _ with the actual editor source cached.

It took like 30 seconds for the job code to load.

My dev console shows this:

```
Job not found 6637fdf1-62f4-446d-8f59-98b7b37dc3e3
```
![Image](https://github.com/user-attachments/assets/be394e89-e618-4f0b-a8d8-da8705bd2034)

That is the job I'm looking at, which _did_ subsequently load.

I think this issue might actually: _sometimes there's an error loading job expressions into the inspector and that stops the editor loading_

**Comment by josephjclark** [comment]
Yes, the editor component will not mount until the workflow has been loaded into the zustand store.

The error in my console was thrown in the editor's mount code:
```js
  render() {
    const { adaptor, disabled, disabledMessage, jobId } = this.el.dataset;

    checkAdaptorVersion(adaptor);

    this.findWorkflowEditorStore().then(workflowStore => {
      const job = workflowStore.getState().getById<Lightning.Job>(jobId);

      if (!job) {
        console.error('Job not found', jobId);
        return;
      }

      if (JobEditorComponent) {
        this.componentRoot?.render(
          <JobEditorComponent
            adaptor={adaptor}
            source={job.body}
            metadata={this.metadata}
            disabled={disabled === 'true'}
            disabledMessage={disabledMessage}
            onSourceChanged={src => this.handleContentChange(src)}
          />
        );
      }
    });
  },
```

**Comment by josephjclark** [comment]
Refreshing that some page sometimes loads instantly, sometimes takes a while. I'm watching a loading spinner as I type this comment.

**Comment by josephjclark** [comment]
Seeing this again today

**Comment by josephjclark** [comment]
Complaints from AK about this today - she's seen it on a demo call

**Comment by rorymckinley** [comment]
@josephjclark @christad92 Could you please give me access to the workflows that you link to in this issue? I think I have picked up some low-hanging fruit on my local dev, but I would like to see the bug in its natural habitat as well.

**Comment by josephjclark** [comment]
@rorymckinley you should be up on https://app.openfn.org/projects/2fedca61-bb42-4e65-a908-7a60f1b25c1e/w/ce0cd63c-4aa1-4e5b-88be-813d125ebb8a?a=2a30533e-0b31-4d5c-ba62-db49de5235ad&s=6637fdf1-62f4-446d-8f59-98b7b37dc3e3&v=72

I don't see anything remarkable about that workflow, but it regularly takes 2-3 seconds to load (obvciously right now it's loading instantly for me, I promise it was slow yesterday). Quite big inputs and outputs I guess? But even then nothing wild

**Comment by rorymckinley** [comment]
> [@rorymckinley](https://github.com/rorymckinley) you should be up on https://app.openfn.org/projects/2fedca61-bb42-4e65-a908-7a60f1b25c1e/w/ce0cd63c-4aa1-4e5b-88be-813d125ebb8a?a=2a30533e-0b31-4d5c-ba62-db49de5235ad&s=6637fdf1-62f4-446d-8f59-98b7b37dc3e3&v=72
> 
> I don't see anything remarkable about that workflow, but it regularly takes 2-3 seconds to load (obvciously right now it's loading instantly for me, I promise it was slow yesterday). Quite big inputs and outputs I guess? But even then nothing wild

Thanks @josephjclark

**Comment by josephjclark** [comment]
We'd added some logging to try and trace occurrences of this.
